src/redux/modules/newsletter.ts

```diff
@@ -127,86 +172,133 @@ const setNewsletters = (
   return { type: SET_NEWSLETTERS, payload: newsletters };
 };
 
+const handleError = (error: ErrorResponse): NewsletterActionTypes => {
+  return {
+    type: ERROR,
+    payload: error,
+  };
+};
+
+export const loading = (): NewsletterActionTypes => {
+  return {
+    type: LOADING,
+    payload: null,
+  };
+};
+
 // Reducer
 const initialState: NewsletterState = {
   newsletters: [] as NewsletterLog[],
   uploadedFile: null,
   uploadStep: 0,
   uploadSelectedTags: [],
+  selectedFilters: [],
+  error: {} as ErrorResponse,
+  loading: false,
 };
 
 export function newsletterReducer(
   state = initialState,
   action: NewsletterActionTypes,
 ): NewsletterState {
   switch (action.type) {
+    case ADD_FILTER:
+      return {
+        ...state,
+        selectedFilters: [...state.selectedFilters, action.payload],
+        loading: false,
+      };
+    case REMOVE_FILTER:
+      return {
+        ...state,
+        selectedFilters: state.selectedFilters.filter(
+          (selectedFilter) => selectedFilter.id !== action.payload.id,
+        ),
+        loading: false,
+      };
+    case SET_NEWSLETTERS:
+      return { ...state, newsletters: action.payload, loading: false };
     case UPLOAD_FILE:
       return {
         ...state,
         uploadedFile: action.payload,
+        loading: false,
       };
     case REMOVE_FILE:
       return {
         ...state,
         uploadedFile: null,
+        loading: false,
       };
     case UPDATE_FILE_UPLOAD_STEP:
       return {
         ...state,
         uploadStep: action.payload,
+        loading: false,
       };
     case ADD_UPLOAD_TAG:
       return {
         ...state,
         uploadSelectedTags: [...state.uploadSelectedTags, action.payload],
+        loading: false,
       };
     case REMOVE_UPLOAD_TAG:
       return {
         ...state,
         uploadSelectedTags: state.uploadSelectedTags.filter(
           (selectedTag) => selectedTag.id !== action.payload.id,
         ),
+        loading: false,
       };
-    case SET_NEWSLETTERS:
-      return { ...state, newsletters: action.payload };
     case ADD_NEWSLETTER:
       return {
         ...state,
         newsletters: [...state.newsletters, action.payload],
+        loading: false,
       };
     case REMOVE_ALL_UPLOAD_TAG:
       return {
         ...state,
         uploadSelectedTags: [],
+        loading: false,
       };
     case ADD_ALL_UPLOAD_TAGS:
       return {
         ...state,
         uploadSelectedTags: action.payload,
+        loading: false,
```

Comment:
think we should try to do smt like this in a fast-follow PR:

https://hackernoon.com/handling-loading-actions-the-proper-way-in-redux-t3k36e8